src/gui/qgsrelationeditorwidget.cpp

```diff
@@ -564,11 +564,17 @@ void QgsRelationEditorWidget::showContextMenu( QgsActionMenu *menu, const QgsFea
   {
     QAction *qAction = nullptr;
 
-    qAction = menu->addAction( QgsApplication::getThemeIcon( QStringLiteral( "/mActionDeleteSelected.svg" ) ),  tr( "Delete Feature" ) );
-    connect( qAction, &QAction::triggered, this, [this, fid]() { deleteFeature( fid ); } );
+    if ( mButtonsVisibility.testFlag( QgsRelationEditorWidget::Button::DeleteChildFeature ) )
```

Comment:
Hm... since we pass the context-menu from `QgsFeatureListView` the menu already contains actions, no?
Couldn't it be that this option would not avoid showing "Delete Feature" and it should be "removed" when not set by the button (same maybe with the duplicateFeature options)?
Or am I missing something?

Reply:
@signedav, I've searched in the code and it seems to me the "Unlink Feature" and "Delete Feature" menu items in the Relation Editor Widget for child features are only added here. I've tested this change in some cases and it seems those menu items are available only when the corresponding buttons are enabled.
Could you please also test the change yourself or suggest some cases in which your concerns could be verified and tested so I can check?

Reply:
I think you are right. "Delete" is not set.

"Duplicate Feature" is set here `QgisApp::setupDuplicateFeaturesAction` why it's still visible.

I'm not sure if the correct approach would be to remove the "duplicate" functions in this menu here or that it should not be set in general on every feature :thinking: